geonode/client/templates/leaflet/maps/map_detail.html

```diff
@@ -460,11 +460,13 @@ <h4>{% trans "Permissions" %}</h4>
         {% include "_permissions_form.html" %}
         {% endif %}
 
+        {% if resource.is_public and "change_resourcebase" in perms_list or resource.layer_group %}
```

Comment:
Brief me on this? Why the resource needs to be public?
Can we create create new map for private map, if we have permission "change_resourcebase".
Also, it's not clear to me what is the precedence of the statement because we have `and` and `or` involved. Maybe put parenthesis to make it easier to read.
Last, `change_resourcebase` doesn't seem to be a straight forward permission. Do we have something like `add_map` permission?

Reply:
cmiiw, `resource.is_public` is a flag for `anyone` in the `who can view it` permissions.
`change_resourcebase` is permission to edit the layer.
`resource.layer_group` is a permission to make sure that the user on the same group with the owner of the layer.
yes, i think i need to add parentheses to make it easy to be read.